boards/samd10-xmini/include/periph_conf.h

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright (C) 2020 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_samd10-xmini
+ * @{
+ *
+ * @file
+ * @brief       Configuration of CPU peripherals for the Atmel SAM D10 Xplained
+ *              Mini board
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#ifndef PERIPH_CONF_H
+#define PERIPH_CONF_H
+
+#include <stdint.h>
+
+#include "cpu.h"
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name   External oscillator and clock configuration
+ *
+ * There are three choices for selection of CORECLOCK:
+ *
+ * - usage of the 48 MHz DFLL fed by external oscillator running at 32 kHz
+ * - usage of the PLL fed by the internal 8MHz oscillator divided by 8
+ * - usage of the internal 8MHz oscillator directly, divided by N if needed
+ *
+ *
+ * The PLL option allows for the usage of a wider frequency range and a more
+ * stable clock with less jitter. This is why this option is default.
+ *
+ * The target frequency is computed from the PLL multiplier and the PLL divisor.
+ * Use the following formula to compute your values:
+ *
+ * CORECLOCK = ((PLL_MUL + 1) * 1MHz) / PLL_DIV
+ *
+ * NOTE: The PLL circuit does not run with less than 32MHz while the maximum PLL
+ *       frequency is 96MHz. So PLL_MULL must be between 31 and 95!
+ *
+ *
+ * The internal Oscillator used directly can lead to a slightly better power
+ * efficiency to the cost of a less stable clock. Use this option when you know
+ * what you are doing! The actual core frequency is adjusted as follows:
+ *
+ * CORECLOCK = 8MHz / DIV
+ *
+ * NOTE: A core clock frequency below 1MHz is not recommended
+ *
+ * @{
+ */
+#define CLOCK_USE_PLL               (1)
+#define CLOCK_USE_XOSC32_DFLL       (0)
+/*
```

Comment:
Wouldn't it be possible to e.g. have three readable constants, like

```C
#define CLOCK_PLL 0
#define CLOCK_XOSC32K_DFLL 1
#define CLOCK_XOSC8M 2

#ifndef USE_CLOCK
#define USE_CLOCK CLOCK_PLL
#endif
```

So that users could add `-DUSE_CLOCK=CLOCK_XOSC23K_DFLL` to modify the clock selection from a Makefile, without having to touch C code?

Reply:
This part is copy & paste from `samd21-xpro`/`samr21-xpro`/`… and should really be moved to a common header.
I can make a separate PR for that.

Reply:
OK